managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2572,6 +2572,17 @@ void internalTrimConsumedLedgers(CompletableFuture<?> promise) {
         internalTrimLedgers(false, promise);
     }
 
+    private OffloadPolicies getOffloadPoliciesIfEnabled() {
+        LedgerOffloader ledgerOffloader = config.getLedgerOffloader();
+        if (ledgerOffloader == null
+                || ledgerOffloader == NullLedgerOffloader.INSTANCE
+                || ledgerOffloader instanceof ReadonlyWrapperLedgerOffloader
```

Comment:
IMO, the readonly should be treat as separate case since you allow readonly to get `OffloadPolicies`

Reply:
Replaced it to `isAppendable()`